### PR TITLE
refactor: consolidate `reqwest` errors

### PR DESCRIPTION
**Pull request by Rolv-Apneseth**
This removes `Error::RequestEncode` and `Error::RequestDecode`. Also did some minor refactoring to remove excessive indentation, and implemented the test for `Error::Reqwest` (though I think maybe that's not necessary - feels like testing implementation details).

Note: I believe I accidentally made this branch from the main repo and not my fork - my bad. If that's an issue I can make a new PR.

### CODE REVIEW

File: src/api/server.rs [hunk 1]
```diff
@@ -538,24 +518,21 @@ impl ServerClient {
...
+            .map_err(Error::Reqwest)?;
+
+        match resp.error_for_status_ref() {
+            Ok(_) => {
+                resp.json::<words::delete::Response>()
+                    .await
+                    .map_err(Error::Reqwest)
             },
-            Err(e) => Err(Error::RequestEncode(e)),
+            Err(_) => Err(Error::InvalidRequest(resp.text().await?)),
```
Comment (jeertmans) [inline src/api/server.rs:535]: I think we should add a test for this.

E.g., sending a request that is too long. I don't know that is the limit of local servers, but the public API has a limit of 1500 characters, I think.

Comment (Rolv-Apneseth) [reply src/api/server.rs:535]: Sure, I'll have a look

Comment (Rolv-Apneseth) [reply src/api/server.rs:535]: Is that last commit there what you had in mind?

Comment (jeertmans) [reply src/api/server.rs:535]: Yes, it looks good @Rolv-Apneseth! Sorry, I didn't see the notification...

Could you fix the conflicts so I can run tests?

Comment (jeertmans) [reply src/api/server.rs:535]: No issue, ping me when you're back ;-)

Comment (Rolv-Apneseth) [reply src/api/server.rs:535]: @jeertmans Did a rebase, hopefully it's all good now

Comment (jeertmans) [reply src/api/server.rs:535]: Perfect, thanks! We will fix CodSpell's warnings later (we need to ignore them)

**Pull request merged**